src/httpClient.ts

```diff
@@ -12,4 +12,19 @@ class RealHttpClient {
   }
 }
 
-export { HttpClient, RealHttpClient };
+class CachingHttpClient {
```

Comment:
Could you add a test case for this please?

Reply:
Done